sdk/test-utils/perfstress/src/recordingClient.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { HttpOperationResponse, URLBuilder } from "@azure/core-http";
+import { DefaultHttpClient, WebResource, WebResourceLike } from "@azure/core-http";
```

Comment:
I understand using core-http is needed to work with storage but it is my understanding that this feature is needed for rate-limited services instead, e.g. text analytics and it uses core v2. How do you feel about making this work with core v2 instead and use it for text analytics?

Reply:
I will get there.
We will eventually end up with a similar solution.
As of now, services depending on core-v2 are not exposing the httpClient option, and same is the case with some services using core-http. 

I'll tweak around to see what it takes to get those libraries onboard. 🙌

As of now, I have this working with the packages that depend on core-http and expose httpClient in the client options.

Reply:
> Even if the latest version of all packages are using core-v2, we want the ability to test older versions using core-v1.

Yes

> I assume there must be some way to implement this, since functional test recording and playback is enabled for all services, correct?

As we talked offline, yes there are ways. Ideal way would be to pass the httpClient or modify the pipeline/policies imo.

Reply:
Here's a quick draft for core-v2 with the addPolicy way https://github.com/HarshaNalluru/azure-sdk-for-js/pull/32 using data-tables as an example.